src/CounterStore.js

```diff
@@ -16,14 +16,28 @@ class CounterStore extends Store {
   }
 
   inc(amount) {
-    const counter = new Counter(this.uid, Object.assign({}, this._index.get()._counters))
+    const counter = new PNCounter(this.uid,
+      new GCounter(this.uid, this._index.get().p._counters),
+      new GCounter(this.uid, this._index.get().n._counters))
```

Comment:
Thanks for highlighting this! ```static from``` in [PNCounter.js](https://github.com/orbitdb/crdts/blob/53207a4b7dd4d1d8f1dff5bebb8d788a77f369b9/src/PN-Counter.js#L41) should be changed to ```static from (json) { 
return new PNCounter(json.id, new GCounter(json.id, json.p.counters), new GCounter(json.id, json.n.counters)) 
} ```

If the above is changed, ```this._index.get().p``` should return a GCounter, that way we can have this be ```const counter = new Counter(this.uid, this._index.get().p, this._index.get().n)```, what do you think?

Reply:
Definitely a better solution as the CRDT lib may be used for something different than OrbitDB libs.

Reply:
I've changed the constructor in [PNCounter](https://github.com/orbitdb/crdts/blob/31f5e8b17acea24cfd8aa2ab157935c724137b7b/src/PN-Counter.js#L8) to expect objects instead of instances of GCounter.
This way this line can be ```const counter = new PNCounter(this.uid, this._index.get().p._counters, this.index.get().n._counters)```, what do you think?